### PR TITLE
feat(index): support SQL to specify inverted index columns

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Allow specifying inverted index columns and support the following scenarios:
1. Avoid using the default strategy of automatically creating an inverted index for the primary key.
2. Manually exclude some primary key columns from indexing.
3. Manually add some non-primary key columns to create indexes.
4. Do not create an inverted index.

SQL statement example:

```sql
CREATE TABLE t (
  "ts" TIMESTAMP,
  "host" STRING,
  "job" STRING,
  "val" DOUBLE,
  TIME INDEX("ts"),
  PRIMARY KEY("host", "job"),
  INVERTED INDEX("job", "val")
);
```

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [x] This PR requires documentation updates.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

**Review by waynexia — APPROVED** [review]
Rest LGTM.

We may consider adding support for specifying this in the "column constrain" position in a follow-up PR

**Comment by killme2008** [comment]
Do we consider the compatibility with the old data?  @zhongzc

**Comment by zhongzc** [comment]
> Do we consider the compatibility with the old data? @zhongzc

Already considered

https://github.com/GreptimeTeam/greptimedb/blob/4a8183f1912948c132f78defdbc46727255a9e13/src/sql/src/statements.rs#L481-L493

File: src/mito2/src/read/scan_region.rs [hunk 1]
```diff
@@ -443,6 +427,45 @@ impl ScanRegion {
...
     }
+
+    fn build_inverted_indexed_column_ids(&self) -> HashSet<ColumnId> {
+        // Default to use primary key columns as inverted index columns.
+        let pk_as_inverted_index = !self
+            .version
+            .metadata
+            .column_metadatas
+            .iter()
+            .any(|c| c.column_schema.has_inverted_index_key());
```
[the remark this answers is not in the document]
Comment (zhongzc) [reply src/mito2/src/read/scan_region.rs:438]: I hope so, but to take care of the old version of column schema (without `invereted_index_key`, the primary key is used by default), the cleanest way I can think of is this.

**Pull request merged**